tests-e2e/core/operation_namespace_test.go

```diff
@@ -0,0 +1,332 @@
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+			ctx = context.Background()
+			if fixture.IsRunningAgainstKCP() {
+				Skip("Skipping this test until we support running gitops operator with KCP")
+			}
+
+			By("creating Opeartion CR")
+			log := log.FromContext(ctx)
+			namespaceToTarget = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: operationNamespace,
+				},
+			}
+			err = k8s.Create(namespaceToTarget, k8sClient)
+			Expect(err).To(Succeed())
+
+			kubeSystemNamespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "kube-system"}}
+			err = k8s.Get(kubeSystemNamespace, k8sClient)
+			Expect(err).To(Succeed())
+
+			dummyApplicationSpec, dummyApplicationSpecString, err := createDummyApplicationData()
+			Expect(err).To(BeNil())
+
+			_, managedEnvironment, _, _, _, err := db.CreateSampleData(dbQueries)
+			Expect(err).To(BeNil())
+
+			gitopsEngineCluster, _, err := dbutil.GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID(ctx, string(kubeSystemNamespace.UID), dbQueries, log)
+			Expect(gitopsEngineCluster).ToNot(BeNil())
+			Expect(err).To(BeNil())
+
+			namespaceToTarget = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: operationNamespace}}
+			err = k8s.Get(namespaceToTarget, k8sClient)
+			Expect(err).To(Succeed())
+
+			By("creating a gitops engine instance")
+			gitopsEngineInstance := &db.GitopsEngineInstance{
+				Gitopsengineinstance_id: "test-engine-instance",
+				Namespace_name:          namespaceToTarget.Name,
+				Namespace_uid:           string(namespaceToTarget.UID),
+				EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+			}
+			err = dbQueries.CreateGitopsEngineInstance(ctx, gitopsEngineInstance)
+			Expect(err).To(BeNil())
+
+			By("Create Application in Database")
+			applicationDB := &db.Application{
+				Application_id:          "test-my-application-id",
+				Name:                    applicationName,
+				Spec_field:              dummyApplicationSpecString,
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			err = dbQueries.CreateApplication(ctx, applicationDB)
+			Expect(err).To(BeNil())
+			operationDB = &db.Operation{
+				Operation_id:            operationName,
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             applicationDB.Application_id,
+				Resource_type:           db.OperationResourceType_Application,
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: "test-user",
+			}
+
+			err = dbQueries.CreateOperation(ctx, operationDB, operationDB.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			operationCR = &managedgitopsv1alpha1.Operation{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      operationName,
+					Namespace: namespaceToTarget.Name,
+				},
+				Spec: managedgitopsv1alpha1.OperationSpec{
+					OperationID: "test-operation",
```

Comment:
Can use `operationDB.Operation_id` here, so that it's obvious what the purpose of this value is, and where this value is coming from.